src/styles/transitions/SlideTransition.js

```diff
@@ -0,0 +1,22 @@
+import BaseTransition, { BaseTransitionPropTypes } from './BaseTransition';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const SlideTransition = ({ direction, children, ...props }) => {
+  return (
+    <BaseTransition classNames={`slide-${direction}`} {...props}>
+      {children}
+    </BaseTransition>
+  );
+};
+
+SlideTransition.defaultProps = {
+  direction: 'right',
+};
+
+SlideTransition.propTypes = {
+  ...BaseTransitionPropTypes,
+  direction: PropTypes.oneOf(['up', 'down', 'right', 'left']),
+};
+
+export default SlideTransition;
```

Comment:
@axbarsan  Maybe I'm missing something here, but id this is just for adding a class to `<BaseTransition>` I really would prefer to keep things simple and just add a css class name.

Is there any other reason to have this?

Reply:
I'd rather leave it like this, in case we ever need to change the implementation of the animation (for example: switching to a full JS animation technique, without any CSS classNames)